cmd/kubelet/app/options/options.go

```diff
@@ -140,6 +165,15 @@ func (f *KubeletFlags) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&f.NodeIP, "node-ip", f.NodeIP, "IP address of the node. If set, kubelet will use this IP address for the node")
 
 	fs.StringVar(&f.ProviderID, "provider-id", f.ProviderID, "Unique identifier for identifying the node in a machine database, i.e cloudprovider")
+
+	fs.StringVar(&f.CertDirectory, "cert-dir", "/var/run/kubernetes", "The directory where the TLS certs are located. "+
+		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
+
+	fs.StringVar(&f.CloudProvider, "cloud-provider", v1alpha1.AutoDetectCloudProvider, "The provider for cloud services. By default, kubelet will attempt to auto-detect the cloud provider. Specify empty string for running with no cloud provider.")
+	fs.StringVar(&f.CloudConfigFile, "cloud-config", "", "The path to the cloud provider configuration file.  Empty string for no configuration file.")
+
+	fs.StringVar(&f.RootDirectory, "root-dir", v1alpha1.DefaultRootDir, "Directory path for managing kubelet files (volume mounts,etc).")
+	fs.StringVar(&f.NodeConfigDir, "node-config-dir", "", "The Kubelet's node configuration controller will use this directory for checkpointing downloaded configurations, finding the (optional) init configuration, and recording miscellaneous information. The path may be absolute or relative; relative paths are under the Kubelet's RootDirectory.")
```

Comment:
please add validation to make sure this isn't set if the dynamic config feature gate is off... don't want people setting this to something weird, it being inert, then coming to life and doing something unexpected in the future